src/tests/ftest/util/server_utils.py

```diff
@@ -525,78 +579,93 @@ def kill(self):
-
+        is_dcpm = self.runner.job.yaml_params.is_scm()
+        is_nvme = self.runner.job.yaml_params.is_nvme()
+        commands = []
+        for server_params in self.runner.job.yaml_params.server_params:
+            scm_list = server_params.scm_list.value
+            scm_mount = server_params.scm_mount.value
+            scm_size = server_params.scm_size.value
+
+            self.log.info("Cleaning up the %s directory.", str(scm_mount))
+
+            # Remove any files left over on each mount point
+            if is_nvme:
+                commands.append("sudo rm -fr {}".format(scm_mount))
+            else:
+                # Also ensure the tmpfs/dcpm mount point exists
+                commands.extend([
+                    "if [ ! -e {} ]".format(scm_mount),
+                    "then sudo mkdir {}".format(scm_mount),
+                    "else find {} -mindepth 1 -maxdepth 1 -print0 | "
+                    "xargs -0r rm -rf".format(scm_mount),
+                    "fi"])
+
+            # Unmount the nvme drive
+            if is_nvme:
+                commands.append("sudo umount {}".format(scm_mount))
+
+            # Clean the dcpm device
+            if is_dcpm:
+                for dcpm_mount in scm_list:
+                    commands.extend([
+                        "sudu umount {}".format(dcpm_mount),
+                        "sudo wipefs -a {}".format(dcpm_mount)])
+
+            # Ensure the tmpfs mount point is mounted
+            if not is_dcpm and not is_nvme:
+                commands.append(
+                    "sudo mount -t tmpfs -o size={}G tmpfs {}".format(
+                        scm_size, scm_mount))
+
+        pcmd(self._hosts, "; ".join(commands), True)
+
+    def storage_prepare(self, user, device_type):
+        """Prepare server's storage using the DAOS server's yaml settings file.
 
-def storage_prepare(hosts, user, device_type):
-    """Prepare storage on servers using the DAOS server's yaml settings file.
+        Args:
+            user (str): username for file permissions
+            device_type (str): storage type - scm or nvme
 
-    Args:
-        hosts (str): a string of comma-separated host names
-        user (str): username for file permissions
-        device_type (str): storage type - scm or nvme
+        Raises:
+            ServerFailed: if server failed to prepare storage
 
-    Raises:
-        ServerFailed: if server failed to prepare storage
+        """
+        # Get the daos_server from the install path. Useful for testing
+        # with daos built binaries.
+        dev_param = ""
+        device_args = ""
+        daos_srv_bin = os.path.join(self.daosbinpath, "daos_server")
+        if device_type == "dcpm":
+            dev_param = "-s"
+        elif device_type == "dcpm_nvme":
+            device_args = " --hugepages=4096"
+        elif device_type == "ram_nvme" or device_type == "nvme":
+            dev_param = "-n"
+            device_args = " --hugepages=4096"
```

Comment:
It's fine to do this explicitly, but I want to point out that it's no longer necessary. As of commit 50e8de, daos_server will do an automatic prepare of the NVMe storage on startup according to the nr_hugepages parameter in the config file. Also, that commit changed the storage prepare invocation to remove the sudo in order to ensure that we're testing request forwarding to daos_admin.

Reply:
Thanks - removed `sudo` from the command.

Reply:
Shouldn't we just totally remove prepare?